### PR TITLE
API Specific major version handling

### LINKED ISSUE #41: Change supported release, recommended major and default major to be API version specific

### PR DESCRIPTION
**Pull request by minorOffense**
Extend the entity properties to support differen major version data per API version.

Fix #41

### CODE REVIEW

**Comment by minorOffense** [comment]
@ClassicCut what's left is writing the hook_update to set the current values for the new properties.

You know the release stuff better than I do so maybe you can take a crack at it. I wrote out what it would need to do. The only part I don't know what to do is to detect the API version (I guess we'd have to scan for release objects and base it on that data?).

**Comment by minorOffense** [comment]
@spotzero Can you look this over too?

**Comment by spotzero** [comment]
Things look good, just missing the update hook, the hard part ;)

File: includes/fserver.entity.inc [hunk 1]
```diff
@@ -172,31 +175,113 @@ function release_uri($entity) {
...
+
+/**
+ * Entity property validation callback for default_major
+ *
+ * @see fserver_project_default_major_property_info().
+ */
+function fserver__project_property_default_major_validate($value) {
+  if (is_array($value)) {
+    foreach ($value as $key => $v) {
+      if (array_key_exists('api', $v) and array_key_exists('default_major', $v)) {
```
Comment (ClassicCut) [inline includes/fserver.entity.inc:232]: @minorOffense Could we not index the array by the API, or would that make validation too tricky?

Comment (minorOffense) [inline includes/fserver.entity.inc:232]: Nope. Entity Property Info for the List type only allows integer keys. Then we get into using a struct by that doesn't work either cause it requires fixed key / value pair definitions.

File: fserver.install [hunk 2]
```diff
@@ -249,3 +251,61 @@ function fserver_update_7000() {
...
+  db_drop_table($temp_table_name);
+  $table = db_create_table($temp_table_name, $schema);
+
+  // load old data into this table
+  $select = db_select('fserver_project', 'proj')
+    ->fields('proj')
+    ->execute();
+  $id = db_insert($temp_table_name)
+    ->fields($fields);
+  while ($record = $select->fetchAssoc()) {
```
Comment (minorOffense) [inline fserver.install:274]: The only question here is can we do this all in one step or should be use batch api to make the update?

**Comment by minorOffense** [comment]
Requires #49 before we can merge. The update script works great. Tested on production dropfort data.

Need #49 so we can update the views to show the information.

**Pull request merged**